src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -50,7 +53,12 @@
     if ([strongDelegate respondsToSelector:@selector(handleQueryImage:completionHandler:)]) {
         if (strongDelegate && mQueue) {
             auto * commandParams = [[MTROtaSoftwareUpdateProviderClusterQueryImageParams alloc] init];
-            ConvertToQueryImageParams(commandData, commandParams);
+            CHIP_ERROR err = ConvertToQueryImageParams(commandData, commandParams);
+            if (err != CHIP_NO_ERROR) {
+                commandObj->AddStatus(commandPath, chip::Protocols::InteractionModel::Status::InvalidCommand);
+                handle.Release();
```

Comment:
Now I wonder if we should move line 48-49 to below this block so we don't need to make an extra `handle.Release` call here. The handler and command path really only needs to be cached before the dispatch below..

Reply:
We could probably do that, yes.